js/src/components/shipping-rate-section/minimum-order-card/minimum-order-card.test.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import MinimumOrderCard from './minimum-order-card.js';
+
+jest.mock( '.~/hooks/useAppSelectDispatch' );
+jest.mock( '.~/hooks/useCountryKeyNameMap' );
+jest.mock( '.~/hooks/useStoreCurrency' );
+
+describe( 'MinimumOrderCard', () => {
+	describe( 'onChange callback property', () => {
+		let value, onChange, rendered;
+		beforeEach( () => {
+			value = Object.freeze( [
+				{
+					id: '1',
+					country: 'US',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {},
+				},
+				{
+					id: '2',
+					country: 'ES',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 20,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+				{
+					id: '3',
+					country: 'CN',
+					method: 'flat_rate',
+					currency: 'USD',
+					rate: 25,
+					options: {
+						free_shipping_threshold: 50,
+					},
+				},
+			] );
+			onChange = jest.fn().mockName( 'onChange callback' );
+
+			rendered = render(
+				<MinimumOrderCard value={ value } onChange={ onChange } />
+			);
+		} );
+
+		afterEach( () => {
+			jest.clearAllMocks();
+		} );
+
+		test( 'When the new minimum order value is provided for a remaining country, calls the `onChange` callback with the new value containing `shippingRate.options.free_shipping_threshold` set to the given value', async () => {
+			// Open "Add another…" modal.
+			userEvent.click( rendered.getByRole( 'button', { name: /Add/ } ) );
+			// Input some value.
+			const input = screen.getByRole( 'textbox' );
+			// For some reason it's typed backwards.
+			await userEvent.type( input, '03' );
```

Comment:
🤯

Reply:
Yup, I was also :exploding_head: 
 I guess it's either a bug in the testing library or in our code that handles the input. But I didn't want to spend more time here, as it is unrelated to this bit.

Reply:
@tomalec , just FYI, apparently the behavior is different in `develop` branch and it causes tests to fail, the test code here has been changed and fixed in PR https://github.com/woocommerce/google-listings-and-ads/pull/1512.

Reply:
:exploding_head: Thanks!